### PR TITLE
Add details about commit type and title structure in the contributor guide.

### PR DESCRIPTION
**Pull request by shulard**
Hello !

I finally started to work on the contributor guide update regarding commit types. I've added some details about the commit title structure and the different commit types.

For me it's enough in this part, maybe I'm wrong. What are you thinking about ?

### CODE REVIEW

**Comment by shulard** [comment]
This PR is related to the RFC hoaproject/Central#61.

File: Contributor/En/Guide.xyl [hunk 1]
```diff
@@ -392,6 +392,40 @@ $ git commit</code></pre>
   <p>Do not edit the <code>CHANGELOG.md</code> file. It is automatically
   generated.</p>
 
+  <h3 id="Commit_message_structure" for="main-toc">Commit message structure</h3>
+
+  <p>Has we already seen in the previous section, a commit title must be 50
```
Comment (Pierozi) [inline Contributor/En/Guide.xyl]: a commit title must be **less than** 50 ?

Comment (shulard) [reply Contributor/En/Guide.xyl]: Yes it's clearer, I've updated the sentence.

File: Contributor/Fr/Guide.xyl [hunk 2]
```diff
@@ -425,6 +425,41 @@ $ git commit</code></pre>
   <p>Ne modifiez pas le fichier <code>CHANGELOG.md</code>. Il est
   généré automatiquement.</p>
 
+  <h3 id="Commit_message_structure" for="main-toc">Structure d'un message de commit</h3>
+
+  <p>Comme nous l'avons vu précédemment, le titre d'un commit fait 50 caractères
+  au maximum. Ce titre se doit d'être précis et sa lecture
+  doit permettre d'identifier la modification apportée.</p>
+
+  <p>Pour faciliter l'écriture de ce titre, nous avons définit une structure
```
Comment (Hywan) [inline Contributor/Fr/Guide.xyl]:
```diff
- nous avons définit
+ nous avons défini
```

File: Contributor/Fr/Guide.xyl [hunk 3]
```diff
@@ -425,6 +425,41 @@ $ git commit</code></pre>
...
+  type : <code>type(scope) title</code><p>
+
+  <p>Cette structure permet de très facilement décrire quel est le type de
```
Comment (Hywan) [inline Contributor/Fr/Guide.xyl]:
```diff
- de très facilement décrire
+ de décrire
```

Never assume something is easy :-).

File: Contributor/Fr/Guide.xyl [hunk 4]
```diff
@@ -425,6 +425,41 @@ $ git commit</code></pre>
...
+  commit et sur quoi il porte. Nous avons en effet isolé plusieurs types
+  récurents dans un projet :</p>
```
Comment (Hywan) [inline Contributor/Fr/Guide.xyl]:
```diff
- récurents
+ récurrents
```

File: Contributor/Fr/Guide.xyl [hunk 5]
```diff
@@ -425,6 +425,41 @@ $ git commit</code></pre>
...
+
+  <ul>
+    <li><code>feat</code> pour une nouvelle fonctionnalité,</li>
```
Comment (Hywan) [inline Contributor/Fr/Guide.xyl]: Use ` ;` (unbreakable space + `;`)  instead of `,` to end items.

File: Contributor/Fr/Guide.xyl [hunk 6]
```diff
@@ -425,6 +425,41 @@ $ git commit</code></pre>
...
+    <li><code>fix</code> pour un correctif,</li>
+    <li><code>test</code> pour une modification liée aux tests,</li>
+    <li><code>doc</code> pour tout ce qui se rapporte à la documentation,</li>
+    <li><code>chore</code> pour tout ce qui concerne la « qualité ».</li>
+  </ul>
+
+  <p>Nous obtenons par exemple :</p>
+
+  <ul>
+    <li><code>feat(permission) Search backward bla bla.,</code></li>
```
Comment (Hywan) [inline Contributor/Fr/Guide.xyl]:
```diff
- bla bla.,</code>
+ bla bla.</code> ;
```

File: Contributor/Fr/Guide.xyl [hunk 7]
```diff
@@ -425,6 +425,41 @@ $ git commit</code></pre>
...
+    <li><code>fix(user) `getName` is incorrectly computing its value.,</code></li>
+    <li><code>…</code></li>
```
Comment (Hywan) [inline Contributor/Fr/Guide.xyl]: You can remove this line I guess :-).

File: Contributor/En/Guide.xyl [hunk 8]
```diff
@@ -392,6 +392,40 @@ $ git commit</code></pre>
   <p>Do not edit the <code>CHANGELOG.md</code> file. It is automatically
   generated.</p>
 
+  <h3 id="Commit_message_structure" for="main-toc">Commit message structure</h3>
+
+  <p>Has we already seen in the previous section, a commit title must be less
```
Comment (Hywan) [inline Contributor/En/Guide.xyl]:
```diff
- Has we already seen
+ As seen
```

File: Contributor/En/Guide.xyl [hunk 9]
```diff
@@ -392,6 +392,40 @@ $ git commit</code></pre>
...
+  than 50 characters long. This title must be precise and reading it must allow
```
Comment (Hywan) [inline Contributor/En/Guide.xyl]:
```diff
- reading it must allow to identify the corresponding modifications.
+ straightfoward regarding the corresponding modifications.
```

File: Contributor/En/Guide.xyl [hunk 10]
```diff
@@ -392,6 +392,40 @@ $ git commit</code></pre>
...
+  to identify the corresponding modifications.</p>
+
+  <p>To facilitate the writing of this title, we defined a structure:
```
Comment (Hywan) [inline Contributor/En/Guide.xyl]:
```
- this title
+ the title
```

```diff
- we defined a structure
+ we expect the following structure
```

File: Contributor/En/Guide.xyl [hunk 11]
```diff
@@ -392,6 +392,40 @@ $ git commit</code></pre>
...
+  <code>type(scope) title</code><p>
```
Comment (Hywan) [inline Contributor/En/Guide.xyl]:
```diff
- </code></p>
+ </code>.</p>
```

File: Contributor/Fr/Guide.xyl [hunk 12]
```diff
@@ -425,6 +425,41 @@ $ git commit</code></pre>
...
   généré automatiquement.</p>
 
+  <h3 id="Commit_message_structure" for="main-toc">Structure d'un message de commit</h3>
+
+  <p>Comme nous l'avons vu précédemment, le titre d'un commit fait 50 caractères
+  au maximum. Ce titre se doit d'être précis et sa lecture
+  doit permettre d'identifier la modification apportée.</p>
+
+  <p>Pour faciliter l'écriture de ce titre, nous avons définit une structure
+  type : <code>type(scope) title</code><p>
```
Comment (Hywan) [inline Contributor/Fr/Guide.xyl]:
```diff
- </code></p>
+ </code>.</p>
```

File: Contributor/En/Guide.xyl [hunk 13]
```diff
@@ -392,6 +392,40 @@ $ git commit</code></pre>
...
+  <p>This structure allow to describe very easily which kind of commit is it and
+  on which part. We have indeed isolated several recurring types in a
+  project:</p>
+
+  <ul>
+    <li><code>feat</code> for a new feature,</li>
+    <li><code>fix</code> for a bug fix,</li>
+    <li><code>test</code> for a test modification,</li>
+    <li><code>doc</code> for a documentation modification,</li>
+    <li><code>chore</code> to replace the “Quality” type.</li>
```
Comment (Hywan) [inline Contributor/En/Guide.xyl]:
```diff
- to replace the “Quality” type
+ to address anything about quality
```

File: Contributor/En/Guide.xyl [hunk 14]
```diff
@@ -392,6 +392,40 @@ $ git commit</code></pre>
...
+  </ul>
+
+  <p>For example, we can obtain:</p>
+
+  <ul>
+    <li><code>feat(permission) Search backward bla bla.,</code></li>
```
Comment (Hywan) [inline Contributor/En/Guide.xyl]:
```diff
- bla.,</code></li>
+ bla.</code>,</li>
```

File: Contributor/En/Guide.xyl [hunk 15]
```diff
@@ -392,6 +392,40 @@ $ git commit</code></pre>
...
+    <li><code>fix(user) `getName` is incorrectly computing its value.,</code></li>
+    <li><code>…</code></li>
```
Comment (Hywan) [inline Contributor/En/Guide.xyl]: You can remove this line.

File: Contributor/En/Guide.xyl [hunk 16]
```diff
@@ -392,6 +392,40 @@ $ git commit</code></pre>
...
+  </ul>
+
+  <p>The goal is to generate a <code>CHANGELOG</code> file as suggested by
+  <a href="http://keepachangelog.com/fr/1.0.0/">http://keepachangelog.com/fr/1.0.0/</a>,
```
Comment (Hywan) [inline Contributor/En/Guide.xyl]: Not sure we want the french link here, prefer http://keepachangelog.com/en/1.0.0/. Also the link text should be “Keep a Changelog”.

File: Contributor/En/Guide.xyl [hunk 17]
```diff
@@ -392,6 +392,40 @@ $ git commit</code></pre>
...
+  so with the “Added”, “Bug fixes”, “Removed“ Sections, and the date of the
+  release.</p>
```
Comment (Hywan) [inline Contributor/En/Guide.xyl]:
```diff
- release
+ releases
```

File: Contributor/Fr/Guide.xyl [hunk 18]
```diff
@@ -425,6 +425,41 @@ $ git commit</code></pre>
...
+
+  <ul>
+    <li><code>feat(permission) Search backward bla bla.,</code></li>
+    <li><code>fix(user) `getName` is incorrectly computing its value.,</code></li>
+    <li><code>…</code></li>
+  </ul>
+
+  <p>L'objectif de la normalisation des messages de commit est de permettre la
+  génération d'un fichier <code>CHANGELOG</code> tel que suggéré sur
+  <a href="http://keepachangelog.com/fr/1.0.0/">http://keepachangelog.com/fr/1.0.0/</a>,
```
Comment (Hywan) [inline Contributor/Fr/Guide.xyl]: Change the link text by “Keep a changelog”.

File: Contributor/Fr/Guide.xyl [hunk 19]
```diff
@@ -425,6 +425,41 @@ $ git commit</code></pre>
...
+  <p>Nous obtenons par exemple :</p>
+
+  <ul>
+    <li><code>feat(permission) Search backward bla bla.,</code></li>
+    <li><code>fix(user) `getName` is incorrectly computing its value.,</code></li>
+    <li><code>…</code></li>
+  </ul>
+
+  <p>L'objectif de la normalisation des messages de commit est de permettre la
+  génération d'un fichier <code>CHANGELOG</code> tel que suggéré sur
```
Comment (Hywan) [inline Contributor/Fr/Guide.xyl]:
```diff
- suggéré sur
+ suggéré par
```

File: Contributor/Fr/Guide.xyl [hunk 20]
```diff
@@ -425,6 +425,41 @@ $ git commit</code></pre>
...
+  <a href="http://keepachangelog.com/fr/1.0.0/">http://keepachangelog.com/fr/1.0.0/</a>,
+  avec des sections « Ajouts », « Correctifs », « Suppressions » ainsi que la date
+  de la sortie de version.</p>
```
Comment (Hywan) [inline Contributor/Fr/Guide.xyl]:
```diff
- de version
+ des versions
```

**Review by Hywan — CHANGES_REQUESTED** [review]
Thanks for the excellent PR!

I've few minor feedbacks. Fix them, and we are ready for a merge :-).

**Comment by shulard** [comment]
Thank you for the feedback, I've updated the files with all the requested changes.

**Comment by shulard** [comment]
Hello @vonglasow, @Hywan, I've checked to be sure that all the review comments have been processed, I think we are ready to go here :smile:.

**Comment by shulard** [comment]
@vonglasow, I've added the changes given by your colleague :smile:. More simple and clearer than the previous version :smile:.

**Review by vonglasow — APPROVED** [review]

**Comment by vonglasow** [comment]
@shulard since this modification is done it looks ok for me too. So I will merge it. If we have more feedback about this section we will create a new PR for that.

**Comment by shulard** [comment]
I've updated the typo, thank for the feedback :smile:.

**Pull request merged**